cluster/tenant-users.go

```diff
@@ -211,3 +216,84 @@ func GetTotalNumberOfUsers(ctx *Context) (int, error) {
 	}
 	return count, nil
 }
+
+// InviteUserByEmail creates a temporary token to signup for service and send an email to the provided user
+func InviteUserByEmail(ctx *Context, user *User) error {
+
+	// generate a token for the email invite
+	// this token expires in 72 hours
+	expires := time.Now().Add(time.Hour * 72)
+
+	urlToken, err := NewURLToken(ctx, &user.ID, TokenSignupEmail, &expires)
+	if err != nil {
+		return err
+	}
+
+	// send email with the invite
+
+	tenant, err := GetTenantWithCtx(ctx, ctx.Tenant.Name)
+	if err != nil {
+		return err
+	}
+
+	// for now, let's hardcode the url, subsequent PRs will introduce system configs
+	signupURL := fmt.Sprintf("http://%s/signup?k=%s", GetInstance().AppURL(), urlToken.String())
+
+	templateData := struct {
+		Name string
+		URL  string
+	}{
+		Name: user.Name,
+		URL:  signupURL,
+	}
+	// Get the mailing template for inviting users
+	body, err := GetTemplate("invite", templateData)
+	if err != nil {
+		return err
+	}
+
+	// send the email
+	err = SendMail(user.Name, user.Email, fmt.Sprintf("Signup for %s Storage", tenant.Name), *body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// setUserPassword sets the password for the provided user by hashing it
+func setUserPassword(ctx *Context, userID *uuid.UUID, password string) error {
+	// validate user Password
+	if password != "" {
+		// TODO: improve regex or use Go validator
+		var re = regexp.MustCompile(`^[a-zA-Z0-9!@#\$%\^&\*]{8,16}$`)
+		if !re.MatchString(password) {
+			return errors.New("a valid password is needed, minimum 8 characters")
+		}
+	}
+	// Hash the password
+	hashedPassword, err := HashPassword(password)
+	if err != nil {
+		return err
+	}
+
+	query := `UPDATE users SET password=$1 WHERE id=$2`
+	tx, err := ctx.TenantTx()
+	if err != nil {
+		return err
+	}
+	stmt, err := tx.Prepare(query)
+	if err != nil {
+		ctx.Rollback()
+		return err
+	}
```

Comment:
[1]Since statements make sense for repeated executions, it makes no sense to use it here because we are also not using stmt.Exec()
```
A prepared statement is a SQL statement with parameter placeholders which is sent to the database server and prepared for repeated execution.
```
https://mindbowser.com/golang-go-database-sql/